alacritty/src/main.rs

```diff
@@ -205,7 +205,14 @@ fn alacritty(options: Options) -> Result<(), Box<dyn Error>> {
     // FIXME patch notify library to have a shutdown method.
     // config_reloader.join().ok();
 
-    // Without explicitly detaching the console cmd won't redraw it's prompt.
+    // Panic on error on Windows, that way the error message is shown in a message box.
+    // It is done before FreeConsole() so the message appears on the console as well.
+    #[cfg(windows)]
+    if let Err(err) = result {
+        panic!("{}", err);
+    }
```

Comment:
I don't like the idea of sidestepping error handling to cause an explicit panic, doesn't really make sense.

Especially not in here, considering there's other places this function could error out from. If anything, our error handling should be consistent?

Reply:
there's only one place and it's an event loop builder, but I think message box won't show if you don't build event loop on windows. Not sure about that though.

It's also tricky that you need to call it before `FreeConsole`, otherwise you won't log into pty, so I don't see other place you'd through it on windows.

The good thing about side stepping panic like that, that it's visible for the users so they can report something at least.

Reply:
That just sounds like the `FreeConsole` code should be reworked.

Reply:
I'm open to suggestions. Ideally, all fatal errors from Alacritty should be reported through GUI on all platforms, as we cannot expect users to run Alacritty from another terminal.

Reply:
The fatal error could be due to crash of the event loop, so it's effectively not possible to do so.

Only windows always has windows around, but not X11/Wayland, you need to somehow draw that sort of thing. In general, you should send information to crash report service, which is system log, basically, so you'll see crashes inside the `journalctl`, etc, since rust's panic doesn't do that sort of thing.

Reply:
You should run `FreeConsole` from inside the `fn main`, from the `match` there you should get a `let result` and then do the same panic logic there.

Reply:
Also, do we really need a `panic`? Can't we call some `Win32` API function with panic message and do a normal return?

Reply:
I've extracted the message box from the panic handler. The message box is called directly without panic.

I'm not sure about moving `FreeConsole` to `main`, but I'll give it a try later.

Reply:
I've removed the GUI and the top-level code from the PR.